### PR TITLE
feat: Expand docs on native backends

### PR DESCRIPTION
**Pull request by loewenheim**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

*Describe your changes here. If your PR relates to or resolves an issue, add a link to that too.*

## Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: src/platforms/native/configuration/backends/index.mdx [hunk 1]
```diff
@@ -11,7 +11,17 @@ The Native SDK can use different backends that are responsible for capturing
...
 CMake option, with support for the following options:
 
-- [`crashpad`](crashpad/), which is the default on Windows and macOS,
-- `breakpad`, which is the default on Linux,
-- `inproc`, the default on Android, and
-- `none`, that disables capturing crashes altogether.
+- [`crashpad`](crashpad/): This uses the out-of-process crashpad handler. It is currently
+  only supported on Desktop OSs, and used as the default on Windows and macOS.
+- `breakpad`: This uses the in-process breakpad handler. It is currently
+  only supported on Desktop OSs, and used as the default on Linux.
```
Comment (Swatinem) [inline src/platforms/native/configuration/backends/index.mdx:17]: I wouldn’t mention "Desktop OSs" at all. crashpad and breakpad work just fine in docker / containers, which are not desktop :-)
Also breakpad "in theory" also works on Android, but we don’t officially support it, primarily because the Java SDK that wraps it is not capable of forwarding minidumps produced by it. And because we don’t have access to all the system symbols on Android.

**Review by Swatinem — APPROVED** [review]

File: src/platforms/native/configuration/backends/index.mdx [hunk 2]
```diff
@@ -11,7 +11,17 @@ The Native SDK can use different backends that are responsible for capturing
...
+- `inproc`: A small in-process handler which is supported on all platforms,
+  and is used as default on Android.
+- `none`: This builds `sentry-native` without a backend, so it does not handle
+  crashes at all. It is primarily used for tests.
+
+`Breakpad` and `inproc` being "in-process" means that they run in the same process
+as the application whose crashes they are meant to capture. Consequently they cannot
+send a crash to Sentry immediately after it happens. Instead, the crash report is
+written to disk and sent the next time the application is run. `Crashpad`, running
+in a separate process, does not suffer from this limitation.
```
Comment (shanamatthews) [inline src/platforms/native/configuration/backends/index.mdx:27]: Great addition, just some tweaks to  simplify the wording:

```suggestion
`Breakpad` and `inproc` both run "in-process", so they run in the same process
as the application they capture crashes for. This means these backends can't
send a crash to Sentry immediately after it happens. Instead, the crash report is
written to disk and sent the next time the application is run. Since `Crashpad` 
runs in a different process it doesn't have this limitation.
```

**Review by shanamatthews — APPROVED** [review]

**Comment by kahest** [comment]
@loewenheim thanks for this :)
